### PR TITLE
Require puppet-agent 6.15.0+ in foreman-installer

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

File: packages/foreman/foreman-installer/foreman-installer.spec [hunk 1]
```diff
@@ -48,7 +48,7 @@ Requires: which
 # puppet-candlepin enables the pki-core module which needs DNF module support
 # That was introduced in 5.5.20 and 6.15.0
 %if 0%{?rhel} == 8
-Requires:   puppet-agent >= 5.5.20
+Requires:   puppet-agent >= 6.1.0
 Conflicts: (puppet-agent >= 6.0.0 with puppet-agent < 6.15.0)
```
Comment (evgeni) [inline packages/foreman/foreman-installer/foreman-installer.spec:52]: Can we fold that into `puppet-agent >= 6.15.0`?

Comment (ehelms) [reply packages/foreman/foreman-installer/foreman-installer.spec:52]: I considered that, but I wanted to be consistent with the testing in foreman-installer.

Comment (ekohl) [reply packages/foreman/foreman-installer/foreman-installer.spec:52]: The way https://github.com/theforeman/puppet-candlepin/pull/200 is written now, it must be `>= 6.15.0` everywhere. With versions older than that it'll always show an error. That's why I wrote https://github.com/theforeman/foreman-installer/pull/683#discussion_r634208321. So IMHO foreman-installer should align with that.

File: packages/foreman/foreman-installer/foreman-installer.spec [hunk 2]
```diff
@@ -18,13 +18,13 @@ Source0:    https://downloads.theforeman.org/%{name}/%{name}-%{version}%{?prerel
 BuildArch:  noarch
 
 Requires:   curl
-Requires:   puppet-agent >= 5.5.0
+Requires:   puppet-agent >= 6.15.0
```
Comment (ehelms) [inline packages/foreman/foreman-installer/foreman-installer.spec:21]: I've now unified this to 6.15.0 across the board given puppet-candlepin needs 6.15.0 as a minimum due to enabling the pki-core module.

**Comment by ehelms** [comment]
This looks like it will need updates proposed in here (the last commit) to pass https://github.com/theforeman/foreman-infra/pull/1610

**Comment by ehelms** [comment]
[test rpm]

**Comment by ehelms** [comment]
I have applied the changes in Koji for external Puppet repositories. This PR is now green building against Puppet 6.

**Review by ekohl — APPROVED** [review]
Thanks! Do we have a Redmine issue to track this?

**Pull request merged**